src/main/java/com/untamedears/JukeAlert/storage/JukeAlertLogger.java

```diff
@@ -671,21 +663,42 @@ public void saveAllSnitches() {
         return info;
     }
 
-    public List<SnitchAction> getSnitchInfo(int snitchId, int offset) {
+    public List<SnitchAction> getSnitchInfo(int snitchId, int offset, LoggedAction filterType, String filterPlayer) {
         List<SnitchAction> info = new ArrayList<SnitchAction>();
-
         try {
-            getSnitchLogStmt.setInt(1, snitchId);
-            getSnitchLogStmt.setInt(2, offset);
-            getSnitchLogStmt.setInt(3, logsPerPage);
-
-            ResultSet set = getSnitchLogStmt.executeQuery();
-            if (set.isBeforeFirst()) {
+            // statement to get LIMIT entries OFFSET from a number from the snitchesDetailsTbl based on a snitch_id from the main snitchesTbl
+            // optionally it can also be filtered by action type and initiated username
+            // LIMIT ?,? means offset followed by max rows to return
+            String stmt = String.format("SELECT * FROM %s WHERE snitch_id=? AND soft_delete = 0", this.snitchDetailsTbl);
```

Comment:
So, this is pretty clever and isn't necessarily _wrong_ but I recommend against "constructing" your query every time someone issues getSnitchInfo. 

You've got 4 queries; enumerate them as private static final strings, and simply pick which string to use base on the filterType and filterPlayer parameters. 

If you think you'll add a lot more in the future and don't want to make them proper static final strings, at the very least, use a StringBuilder here explicitly instead of string concatenation.